### ISSUE TITLE
Paths configuration prefixes regular modules with specified path

### ISSUE THREAD

**Issue by segersniels** [swc-project/swc#8375]
### Describe the bug

For some reason if you specify a `baseUrl` and `paths` config like you would otherwise in `tsconfig`, SWC decides to try to resolve regular `node_modules` from the specified paths.

### Input code

```typescript
import { helloWorld } from "hello";
import * as Sentry from '@sentry/node';

Sentry;
helloWorld("SWC");
```

### Config

```json
{
  "inlineSourcesContent": true,
  "jsc": {
    "keepClassNames": true,
    "output": {
      "preamble": ""
    },
    "parser": {
      "syntax": "typescript"
    },
    "baseUrl": "/Users/nielssegers/personal/swc-paths/",
    "paths": {
      "*": [
        "src/*"
      ]
    },
    "preserveAllComments": true,
    "target": "es2022",
    "transform": {
      "decoratorMetadata": false,
      "legacyDecorator": false
    }
  },
  "minify": false,
  "module": {
    "allowTopLevelThis": false,
    "ignoreDynamic": false,
    "lazy": false,
    "noInterop": false,
    "preserveImportMeta": false,
    "resolveFully": false,
    "strict": false,
    "strictMode": true,
    "type": "commonjs"
  },
  "sourceMaps": true
}
```

### Playground link (or link to the minimal reproduction)

https://github.com/segersniels/swc-paths-issue

### SWC Info output

```
npx -y swc-info@latest

    Operating System:
        Platform: darwin
        Arch: arm64
        Machine Type: arm64
        Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:12 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T8103
        CPU: (8 cores)
            Models: Apple M1

    Binaries:
        Node: 20.9.0
        npm: 10.1.0
        Yarn: 1.22.21
        pnpm: N/A

    Relevant Packages:
        @swc/core: 1.3.100
        @swc/helpers: N/A
        @swc/types: 0.1.5


    SWC Config:
        output: N/A
        .swcrc path: N/A

    Next.js info:
        output: N/A
```

### Expected behavior

I would expect `swc` to correctly differentiate between a local file and an actual module from `node_modules` that node needs to resolve.

### Actual behavior

Instead of leaving a regular module `@sentry/node` alone, `swc` decided to add the `src/` prefix to the output which stops it from resolving since it thinks it's a local file.

Run `npm run build` in the example and look at the output of `dist/index.js`.

```ts
const _hello = require("./src/hello");
const _node = _interop_require_wildcard(require("src/@sentry/node"));
```

### Version

0.1.63

**Comment by segersniels** [comment]
For now I'm just working around it by moving the entrypoint of the application to the `src` directory and changing the `paths` configuration to:

```json
{
  "compilerOptions": {
    "baseUrl": "./src",
    "paths": {
      "*": ["*"]
    }
  }
}
```

That way the `src` prefix can be removed.

**Comment by segersniels** [comment]
I see that the issue is seemingly fixed. When is the fix expected to release? I have tried with the latest versions of both the crate and npm and the issue still seems to be present.

**Comment by kdy1** [comment]
I think it's not fixed correctly, sorry

**Comment by segersniels** [comment]
> I think it's not fixed correctly, sorry

No worries, here to assist if you need more information!

**Comment by segersniels** [comment]
@kdy1 Is the fix included in the latest published version? Updated the playground example to the latest version but issue is still present.

**Comment by kdy1** [comment]
Nope, but it should be included in [1.3.106-nightly-20240122.2](https://github.com/swc-project/swc/commit/240ed1385df65e4711c7455340242fee29a126aa)
